client/verta/tests/unit_tests/strategies.py

```diff
@@ -262,3 +260,98 @@ def mock_workspace(draw):
         )
     )
     return workspace
+
+
+@st.composite
+def mock_pipeline_definition(draw):
+    """Generate a mocked pipeline specification dictionary"""
+
+    # step names in a pipeline must be unique
+    step_names = draw(
+        st.lists(st.text(min_size=5, max_size=25), min_size=5, max_size=5, unique=True)
```

Comment:
Did you have a reason in mind for `min_size=5`? For property testing, it's better to set technical limits (like `st.text(min_size=1)` and `st.lists(min_size=2)` rather than convenient/typical limits so we don't miss edge cases.

Reply:
We could dynamically build the `graph` from any arbitrary number of steps now that we are just using a straight linear pipeline.  (Was using something more complex before, but abandoned that as it isn't part of the vision for v1 any way, so there is no longer any need to hand-roll the graph).  I'll take a stab at something that builds `graph` for any arbitrary number of steps

Reply:
Done